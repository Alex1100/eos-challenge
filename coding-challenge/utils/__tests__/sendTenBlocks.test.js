const { sendTenBlocks,  } = require('../sendTenBlocks');
const { lastTenBlockPropsWithoutTypeName } = require('../fixtures/lastTenBlockProps');
const { dummyData } = require('../fixtures/blockListStreamProps');

describe('sendTenBlocks', () => {
  const result = lastTenBlockPropsWithoutTypeName.getLastTenBlocks;

  it('tests the sendTenBlocks function', () => {
    expect(sendTenBlocks(dummyData)).toEqual(result);
    expect(sendTenBlocks(dummyData)).toMatchSnapshot();
  });
});
