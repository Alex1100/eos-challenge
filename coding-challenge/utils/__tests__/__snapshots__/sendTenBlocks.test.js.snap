// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sendTenBlocks tests the sendTenBlocks function 1`] = `
Array [
  Object {
    "action_mroot": "eca7566e358a738c88d79260000cc1a33fb886241e4513035fdb39cf93821207",
    "block_mroot": "6958430f58ab70a27033126a195e74adeaf9afeabc926923ae2d209cdf2f2c21",
    "block_num": 6074,
    "error": "",
    "input_transactions": 0,
    "previous": "000017b95dbc82ee20eaa05970918be17f5dbdddd3948c4df8703cd76b4c58ab",
    "producer": "eosio",
    "producer_signature": "EOSKkogD1VH5H3t6DgsVYW7AxiU8wM5N5GJyKvzVQniqWWmutWDCLzxdVTuYNyPEsFpEEhjVC4diFwQbe5dDfcp2D4cXxwCw1",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "d008d61b274b94c9ab5df43c3c5d6d54bf9af690c8371ce5e5fdd8e3f62cf215",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:05.500",
    "transaction_mroot": "be099291c3536fade8041b53483b50ac3ca708611c55bab555878e26e50a7f2c",
  },
  Object {
    "action_mroot": "09d3dc781f4ee7a9668443e9523424a03d950e99dd2a617cbb8c7b596ba11612",
    "block_mroot": "b8710a244b93a2b8d07d6d685d32426bd0a104f95339f714c3d31ba37d9f729e",
    "block_num": 6075,
    "error": "",
    "input_transactions": 0,
    "previous": "000017bab15515bdbbbc277d58e51a01839255ffbab5157a43ae431bb3d82302",
    "producer": "eosio",
    "producer_signature": "EOSJyvVx8rYnZ9pvWDjjkZnswQZa95d5hvrsNtzsydMXjWj9ZCRso7Vt7awagjQ8x5yqMYCEcNbTzRKQegRBDZmVgquytHxuf",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "ddf3688c082d089e0467a3e9168f7062407fa924adbbe3c222b910d9f60f4fe8",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:06.000",
    "transaction_mroot": "97236423032e50fb547c26779f658fc3316b69c06e62b35cb01ab9b6fef98867",
  },
  Object {
    "action_mroot": "8c3bf3dfa7bcd9749982347afccf2a0d0e8ec377d6720917bb0fa8ae522bf2dc",
    "block_mroot": "899f025114d84c57bf58a6c9627e17177a7a116dcd7eabca7afd440b40aa9aa6",
    "block_num": 6076,
    "error": "",
    "input_transactions": 0,
    "previous": "000017bb412b1071535e510237349695fc7e351797936c872cbef8fce9a1804b",
    "producer": "eosio",
    "producer_signature": "EOSKbYGPWTzGpFR17Ww2LXW1uHwHF8DSKZZd3ns3NMCAbn2pLb5ibZUTz6kUDbatscnsV9Gp5QfU2LuEEr31hT9iJrdRZq55T",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "76ebbd2e7a60549487df105f622088e7fc347e0ccd38dab2eda6500ff5ddd11b",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:06.500",
    "transaction_mroot": "9bf457ced0499174d4c977e3701db4293b543e13f75ea2514d0a6b0d3b3b9410",
  },
  Object {
    "action_mroot": "a5d1bbd633a29a96723a9c89c90534d422321533915d4968bbaf5ceadbdff67f",
    "block_mroot": "77d7f2678beceb46e05a7ea432c1844ab10bba3716e82c090516242881e1c51b",
    "block_num": 6077,
    "error": "",
    "input_transactions": 0,
    "previous": "000017bc3865555faa82e4403322330c3e14ee402137a222d25544f015c72d7d",
    "producer": "eosio",
    "producer_signature": "EOSK7BmMToum1KGg7aUM3EAoKaEbU1Av4j1mL3cnUN1vGZ5iUYR5BU6XocEGzyctH7wDAVdYC9oehz6LQMvSoRi7K8NzFK7pC",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "edc4c60c1f4d3d2e4487169ce802abc0f8c8836a96b1029f9ed73cfd55f8d2b2",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:07.000",
    "transaction_mroot": "268f11b2e732c63b84a54c78c562764d8867d7d67e50a3b1cfd20a4b6c84d2ab",
  },
  Object {
    "action_mroot": "3151f690d87ed83a9ed5f277dfd7788f48c96ec5363997c68992d2024dcd98dd",
    "block_mroot": "2c9b0c1bccbdeb87d28beb1a01c3cd47d62279fc007659dba8a6f207f2efaf28",
    "block_num": 6078,
    "error": "",
    "input_transactions": 0,
    "previous": "000017bd7ceb45dd7c36833ccf7311882460360f87ea5547362bb4eda341ae9f",
    "producer": "eosio",
    "producer_signature": "EOSK811mbYx1dAnMFmZ9Yt7BYhbZbq65DxoscG8jUj2NzHHk5yfT4thnZPs42NSgKyHG6vkrLjjeHu2BnDKQvM9b3WUAc91R7",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "b07ad193e59565245e50a788a9c672abd3dbbdd0ddeb80d60ab973c6a2616b42",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:07.500",
    "transaction_mroot": "c44fbf4b3b7041f23552d1fdd01f6c4ba5b7ca1cadc33038711749e8554df48c",
  },
  Object {
    "action_mroot": "7bda78f9b5413a344670139bd07c7fcf6ef7ef215de2159bf70245436e7e4fa1",
    "block_mroot": "096f6edd2eff98d2e33912ba9d936c28bf7ec83bc56aa831ad2d54dcc27b0b46",
    "block_num": 6079,
    "error": "",
    "input_transactions": 0,
    "previous": "000017be85a17d42979b5263439bda8945786d39b3f75a32074630f4b4cd4e55",
    "producer": "eosio",
    "producer_signature": "EOSJyQtfs6JpUeAbGGxhWVZokz8zG2DrM1oUXB2axi5WbtcGuxcfsk9BXzqtL5c5ezXL7bDQoDSsdjBPgRPQGiHmikxu2gXcj",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "f65f2de3f02a2eba9b89590be908600011c6d479c97e1c69356ee6bc55df194e",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:08.000",
    "transaction_mroot": "c7cd7d3da278743abd1228a65b87f9e9f8513ee2d9cc555021ab5a8036029faf",
  },
  Object {
    "action_mroot": "51ac8a0c2080bbccb0f5464bf50e4ff882d0f99331b6a7ac4951e3ba55e3b262",
    "block_mroot": "6289f8b3d6150b1e76ee02f894447a8537ab6e45c764649e12c668de5b6c97ce",
    "block_num": 6080,
    "error": "",
    "input_transactions": 0,
    "previous": "000017bf8826a8b6b887fd963362042fa49b8d04fe206d8cec94760dc151250b",
    "producer": "eosio",
    "producer_signature": "EOSKgcF4b21sSRhpkfJ61zgGeZV6NaGwAJdZ6KhbtcfvVgEXjH3B9LPgPnAjbxcq4J6nxeRWb8Xwoxmxmk9eTeghq7cCJjKC4",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "690f885cbb8f74836e3449c6a52e249342e1725e8b6be89e150c74f8ad7287f2",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:08.500",
    "transaction_mroot": "dc68a49ab92c7d75ebd661a00a3e0fd7445e8a410d970d0f23c9b17db9a7608a",
  },
  Object {
    "action_mroot": "1b7a24cb4bc604a4961cda7461da675f4b09eccc4f4fba1949bda91ae096eca0",
    "block_mroot": "e7d3da9511507c87d18942b11ec510ad3843fb8913dd7042893529c20259c470",
    "block_num": 6081,
    "error": "",
    "input_transactions": 0,
    "previous": "000017c018812cb416724239094f7306230bd920a4e02670a00ac0a3d99fce87",
    "producer": "eosio",
    "producer_signature": "EOSKae6ze2ugBDSb8J3tHCrqGxYrHNxKpxHSEHKGFvXMDULPUov3JbmATzAqjW3FCTw48PerW2onNkXUx9kfwHSf9HJsjCZqs",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "889f524fb983c643471e97ea927a80b314ac2b0957a00634d3153835f8f0ef39",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:09.000",
    "transaction_mroot": "77553c705b417c46ca709c6831557509480331e5cb33b96bac4c5edd8bd3a94c",
  },
  Object {
    "action_mroot": "a7f8402872a72c0549ff51b879de17832a56601714d3c24899dc6cbb82672848",
    "block_mroot": "b1cb4602c2046c0a4e6284e8c3d766affd948f0b36ee84eed10e347445139932",
    "block_num": 6082,
    "error": "",
    "input_transactions": 0,
    "previous": "000017c12f87a7431387b393bd4a851b7668c52d1f1e85a402a7188c81facbb9",
    "producer": "eosio",
    "producer_signature": "EOSKZ3SQXcTvYQYJMgoHipR53K2iWmLp7uD94gE2i6nXU82KfLJ9yKSLwEMzcRai6qxJFhqwjFB5yVsjaKRUc8CettnehiDMH",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "854ff136a82ed26aac544dcbb4def3bd39cf3a072b3bb0d2cdd3aad6d7505e33",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:09.500",
    "transaction_mroot": "e15950b1d477ce3bf5969c29bbbc731e840ee443c67a6fa925bcaa012b807009",
  },
  Object {
    "action_mroot": "57c7f7c6a73d9d6ae2d3895a25370e92a47d69a0f4d281e5adc3e6ac84a4242d",
    "block_mroot": "d487fa1f8f3ce0921d16833aab5002590cda168e6bfb52e1caddcc6bd5caa285",
    "block_num": 6083,
    "error": "",
    "input_transactions": 0,
    "previous": "000017c210a901c843ddca0cb12d76b06eb8c9562bd93f43a22fdc6e8c466748",
    "producer": "eosio",
    "producer_signature": "EOSJzFgtqNoycFrc7pjQaPvJS8UqAN5ghiFenLqRJGaks3HqbTEMVJFoh9xXcWvt1XaA7TkVgRX5yXnv7W25e5vpAisrkrspR",
    "regions": Object {
      "cycles_summary": Object {
        "read_locks": Array [],
        "transactions": Object {
          "id": "7193c6784b3694950453a79e15770bd60e173b497096b231e380937be0deaafb",
          "kcpu_usage": 2,
          "net_usage_words": 38,
          "status": "executed",
        },
        "write_locks": Array [],
      },
      "region": 0,
    },
    "schedule_version": 0,
    "timestamp": "2018-04-26T01:26:10.000",
    "transaction_mroot": "703b28e5664099c02a9311e34b2795e359d2f624247407134b374c99f36eefcd",
  },
]
`;
